controllers/dexserver_controller.go

```diff
@@ -381,6 +381,17 @@ func getConnectorSecretFromRef(connector authv1alpha1.ConnectorSpec, m *authv1al
 		}
 		checkAndAddLabelToSecret(resource, r, ctx)
 		return string(resource.Data["bindPW"]), nil
+	case authv1alpha1.ConnectorTypeOIDC:
+		secretName = connector.OIDC.ClientSecretRef.Name
+		if secretNamespace = connector.OIDC.ClientSecretRef.Namespace; secretNamespace == "" {
```

Comment:
Should/can we refactor this given that the code looks the same for all connector types other than secret name and which data from the secret is returned?

Reply:
Secret name is different for each connector. Even if I refactor this logic, still I'll have to add switch for secret name. I think we should keep it as it is. thoughts?

Reply:
I see the check for namespace is different in each case as well, so maybe not quite to the point that we need to refactor.